lib/dns.js

```diff
@@ -237,6 +255,16 @@ function lookup(hostname, options, callback) {
   return req;
 }
 
+// DNS cache checks every 1 second
+setInterval(() => {
```

Comment:
I'm not a huge fan of a setInterval running when the dns cache is empty. If this route is approved, I suggest a setTimeout that gets called if the cache was empty before a value was added, and setTimeout gets called again if the dnsCache is empty after the interval executed.

Also, the interval/timeout itself should also be unrefed.